crates/wal-protocol/src/timer.rs

```diff
@@ -48,21 +48,30 @@ impl TimerValue {
     pub fn new_invoke(
         invocation_id: InvocationId,
         wake_up_time: MillisSinceEpoch,
-        entry_index: EntryIndex,
         service_invocation: ServiceInvocation,
     ) -> Self {
-        let timer_key = TimerKey {
-            invocation_uuid: invocation_id.invocation_uuid(),
-            timestamp: wake_up_time.as_u64(),
-            journal_index: entry_index,
-        };
+        let timer_key =
+            TimerKey::new_invocation(wake_up_time.as_u64(), invocation_id.invocation_uuid());
 
         Self {
             timer_key,
             value: Timer::Invoke(service_invocation),
         }
     }
 
+    pub fn new_clean_invocation_status(
+        invocation_id: InvocationId,
+        wake_up_time: MillisSinceEpoch,
+    ) -> Self {
+        TimerValue {
+            timer_key: TimerKey::new_invocation(
+                wake_up_time.as_u64(),
+                invocation_id.invocation_uuid(),
+            ),
```

Comment:
I think this should be another timer kind, specific for the action of cleaning invocation status. Imagine we'll be able to support later cleaning up at separate times the invocation status and the idempotency key.

Reply:
What you are describing should already be supported. The one thing that is not supported is to have multiple actions scheduled at the same time for a single invocation because they would have the same key. Currently, we don't have this requirement. Once we have it, then we can introduce a new `TimerKind` for the concurrent action.

Maybe I should have called `TimerKind` `TimerScope` instead.

Reply:
I guess the argument one could make is that having a `TimerKind` per `Timer` would establish a one to one mapping between these two types.

Reply:
> I guess the argument one could make is that having a TimerKind per Timer would establish a one to one mapping between these two types.

Yes this is also true, i think it is overall simpler to understand.

> The one thing that is not supported is to have multiple actions scheduled at the same time for a single invocation because they would have the same key. Currently, we don't have this requirement. Once we have it, then we can introduce a new TimerKind for the concurrent action.

Not yet, but as described above, we already have some ideas in mind where multiple timers with same scope would be required, so I think it makes sense from now to support that.